### ISSUE TITLE
keadm init: "Error: the server could not find the requested resource"

### ISSUE THREAD

**Issue by BenAsaf** [kubeedge/kubeedge#2988]
Hello, 
First time users for KubeEdge and we're pretty much stuck at the beginning. Seems like there is a missing step in the Installation?

We tried to use this documentation: https://kubeedge.io/en/docs/setup/keadm/
And then we found this one: https://kubeedge-docs.readthedocs.io/en/latest/setup/setup.html

We didn't get to installing on Edge Device / Nodes because installation of Master fails.

Any guidance, tips, insights will be highly appreciated! 

**What happened**:
1. We setup Kubernetes Cluster on EC2 Instance
2. Downloaded the 3 binaries versioned 1.7.1 from Releases (edgesite, keadm, kubeedge)
3. Ran the command: 
`sudo ./keadm init --kube-config=/home/ubuntu/.kube/config --advertise-address="Public-IP-Here"`

Got the error message:
```
ubuntu@Kube-Master01:~/kubeedge/kubeedge1.7.1/_output/local/bin$ ./keadm init --kube-config=/home/ubuntu/.kube/config
Kubernetes version verification passed, KubeEdge installation will start...
Error: the server could not find the requested resource
Usage:
  keadm init [flags]

Examples:

keadm init

- This command will download and install the default version of KubeEdge cloud component

keadm init --kubeedge-version=1.5.0  --kube-config=/root/.kube/config

  - kube-config is the absolute path of kubeconfig which used to secure connectivity between cloudcore and kube-apiserver


Flags:
      --advertise-address string   Use this key to set IPs in cloudcore's certificate SubAltNames field. eg: 10.10.102.78,10.10.102.79
      --domainname string          Use this key to set domain names in cloudcore's certificate SubAltNames field. eg: www.cloudcore.cn,www.kubeedge.cn
  -h, --help                       help for init
      --kube-config string         Use this key to set kube-config path, eg: $HOME/.kube/config (default "/root/.kube/config")
      --kubeedge-version string    Use this key to download and use the required KubeEdge version
      --master string              Use this key to set K8s master address, eg: http://127.0.0.1:8080
      --tarballpath string         Use this key to set the temp directory path for KubeEdge tarball, if not exist, download it

F0731 17:18:10.401847   27409 keadm.go:27] the server could not find the requested resource
goroutine 1 [running]:
k8s.io/klog/v2.stacks(0xc000138001, 0xc000170210, 0x5c, 0xb0)
	/home/ubuntu/kubeedge/kubeedge1.7.1/vendor/k8s.io/klog/v2/klog.go:996 +0xb9
k8s.io/klog/v2.(*loggingT).output(0x2bd1580, 0xc000000003, 0x0, 0x0, 0xc0008776c0, 0x21b24dd, 0x8, 0x1b, 0x1db3e00)
	/home/ubuntu/kubeedge/kubeedge1.7.1/vendor/k8s.io/klog/v2/klog.go:945 +0x191
k8s.io/klog/v2.(*loggingT).printDepth(0x2bd1580, 0xc000000003, 0x0, 0x0, 0x1, 0xc00079ff68, 0x1, 0x1)
	/home/ubuntu/kubeedge/kubeedge1.7.1/vendor/k8s.io/klog/v2/klog.go:718 +0x165
k8s.io/klog/v2.(*loggingT).print(...)
	/home/ubuntu/kubeedge/kubeedge1.7.1/vendor/k8s.io/klog/v2/klog.go:703
k8s.io/klog/v2.Fatal(...)
	/home/ubuntu/kubeedge/kubeedge1.7.1/vendor/k8s.io/klog/v2/klog.go:1436
main.main()
	/home/ubuntu/kubeedge/kubeedge1.7.1/keadm/cmd/keadm/keadm.go:27 +0xa6

goroutine 19 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x2bd1580)
	/home/ubuntu/kubeedge/kubeedge1.7.1/vendor/k8s.io/klog/v2/klog.go:1131 +0x8b
created by k8s.io/klog/v2.init.0
	/home/ubuntu/kubeedge/kubeedge1.7.1/vendor/k8s.io/klog/v2/klog.go:416 +0xd8

goroutine 28 [runnable]:
text/template/parse.(*lexer).emit(...)
	/snap/go/7954/src/text/template/parse/lex.go:157
text/template/parse.lexText(0xc00016a800, 0x1c60b98)
	/snap/go/7954/src/text/template/parse/lex.go:269 +0x4fd
text/template/parse.(*lexer).run(0xc00016a800)
	/snap/go/7954/src/text/template/parse/lex.go:230 +0x37
created by text/template/parse.lex
	/snap/go/7954/src/text/template/parse/lex.go:223 +0x14b

goroutine 27 [runnable]:
golang.org/x/net/http2.(*clientStream).awaitRequestCancel(0xc0007ab4a0, 0xc0007c2f00)
	/home/ubuntu/kubeedge/kubeedge1.7.1/vendor/golang.org/x/net/http2/transport.go:343
created by golang.org/x/net/http2.(*clientConnReadLoop).handleResponse
	/home/ubuntu/kubeedge/kubeedge1.7.1/vendor/golang.org/x/net/http2/transport.go:2054 +0x728

goroutine 34 [IO wait]:
internal/poll.runtime_pollWait(0x7fcb731cb888, 0x72, 0xffffffffffffffff)
	/snap/go/7954/src/runtime/netpoll.go:222 +0x55
internal/poll.(*pollDesc).wait(0xc000575098, 0x72, 0x700, 0x7b5, 0xffffffffffffffff)
	/snap/go/7954/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
	/snap/go/7954/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Read(0xc000575080, 0xc000194800, 0x7b5, 0x7b5, 0x0, 0x0, 0x0)
	/snap/go/7954/src/internal/poll/fd_unix.go:166 +0x1d5
net.(*netFD).Read(0xc000575080, 0xc000194800, 0x7b5, 0x7b5, 0x77e, 0xc0002661a0, 0xd)
	/snap/go/7954/src/net/fd_posix.go:55 +0x4f
net.(*conn).Read(0xc000010e88, 0xc000194800, 0x7b5, 0x7b5, 0x0, 0x0, 0x0)
	/snap/go/7954/src/net/net.go:183 +0x91
crypto/tls.(*atLeastReader).Read(0xc000123c80, 0xc000194800, 0x7b5, 0x7b5, 0x77e, 0xc000105800, 0x0)
	/snap/go/7954/src/crypto/tls/conn.go:776 +0x63
bytes.(*Buffer).ReadFrom(0xc000266278, 0x1db28e0, 0xc000123c80, 0x419925, 0x19ab020, 0x1b1a520)
	/snap/go/7954/src/bytes/buffer.go:204 +0xbe
crypto/tls.(*Conn).readFromUntil(0xc000266000, 0x1db4400, 0xc000010e88, 0x5, 0xc000010e88, 0xd)
	/snap/go/7954/src/crypto/tls/conn.go:798 +0xf3
crypto/tls.(*Conn).readRecordOrCCS(0xc000266000, 0x0, 0x0, 0x921c1e)
	/snap/go/7954/src/crypto/tls/conn.go:605 +0x115
crypto/tls.(*Conn).readRecord(...)
	/snap/go/7954/src/crypto/tls/conn.go:573
crypto/tls.(*Conn).Read(0xc000266000, 0xc000618000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/snap/go/7954/src/crypto/tls/conn.go:1276 +0x165
bufio.(*Reader).Read(0xc0006040c0, 0xc0000fe2d8, 0x9, 0x9, 0x0, 0x10, 0x7fcb7332f5b8)
	/snap/go/7954/src/bufio/bufio.go:227 +0x222
io.ReadAtLeast(0x1db2740, 0xc0006040c0, 0xc0000fe2d8, 0x9, 0x9, 0x9, 0xc00064fcd0, 0x9094bc, 0x1a5ff00)
	/snap/go/7954/src/io/io.go:328 +0x87
io.ReadFull(...)
	/snap/go/7954/src/io/io.go:347
golang.org/x/net/http2.readFrameHeader(0xc0000fe2d8, 0x9, 0x9, 0x1db2740, 0xc0006040c0, 0x0, 0x0, 0xc00042c0b0, 0x0)
	/home/ubuntu/kubeedge/kubeedge1.7.1/vendor/golang.org/x/net/http2/frame.go:237 +0x89
golang.org/x/net/http2.(*Framer).ReadFrame(0xc0000fe2a0, 0xc00042c0b0, 0x0, 0x0, 0x0)
	/home/ubuntu/kubeedge/kubeedge1.7.1/vendor/golang.org/x/net/http2/frame.go:492 +0xa5
golang.org/x/net/http2.(*clientConnReadLoop).run(0xc00064ffa8, 0x0, 0x0)
	/home/ubuntu/kubeedge/kubeedge1.7.1/vendor/golang.org/x/net/http2/transport.go:1819 +0xd8
golang.org/x/net/http2.(*ClientConn).readLoop(0xc000606000)
	/home/ubuntu/kubeedge/kubeedge1.7.1/vendor/golang.org/x/net/http2/transport.go:1741 +0x6f
created by golang.org/x/net/http2.(*Transport).newClientConn
	/home/ubuntu/kubeedge/kubeedge1.7.1/vendor/golang.org/x/net/http2/transport.go:705 +0x6c5
```

**What you expected to happen**:
We were expecting the following output, as stated in Docs:
```
Output:

Kubernetes version verification passed, KubeEdge installation will start...
...
KubeEdge cloudcore is running, For logs visit:  /var/log/kubeedge/cloudcore.log
```

**How to reproduce it (as minimally and precisely as possible)**:
1. Setup EC2 cluster
2. Follow the docs https://kubeedge.io/en/docs/setup/keadm/
**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
`1.15`
- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`):
1.7.1 through the Releases binaries. We were still setting up the Cloud-Side.
- <details><summary>Cloud nodes Environment:</summary>

  - Hardware configuration (e.g. `lscpu`):
   ```
   Architecture:        x86_64
   CPU op-mode(s):      32-bit, 64-bit
   Byte Order:          Little Endian
   CPU(s):              2
   On-line CPU(s) list: 0,1
   Thread(s) per core:  1
   Core(s) per socket:  2
   Socket(s):           1
   NUMA node(s):        1
   Vendor ID:           GenuineIntel
   CPU family:          6
   Model:               63
   Model name:          Intel(R) Xeon(R) CPU E5-2676 v3 @ 2.40GHz
   Stepping:            2
   CPU MHz:             2399.828
   BogoMIPS:            4800.07
   Hypervisor vendor:   Xen
   Virtualization type: full
   L1d cache:           32K
   L1i cache:           32K
   L2 cache:            256K
   L3 cache:            30720K
   NUMA node0 CPU(s):   0,1
   Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht    syscall nx rdtscp lm constant_tsc rep_good nopl xtopology cpuid pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm cpuid_fault invpcid_single pti fsgsbase bmi1 avx2 smep bmi2 erms invpcid xsaveopt
  ```
  - OS (e.g. `cat /etc/os-release`):
   ```
   ubuntu@Kube-Master01:~$ cat /etc/os-release
   NAME="Ubuntu"
   VERSION="18.04.5 LTS (Bionic Beaver)"
   ID=ubuntu
   ID_LIKE=debian
   PRETTY_NAME="Ubuntu 18.04.5 LTS"
   VERSION_ID="18.04"
   HOME_URL="https://www.ubuntu.com/"
   SUPPORT_URL="https://help.ubuntu.com/"
   BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
   PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
   VERSION_CODENAME=bionic
   UBUNTU_CODENAME=bionic
   ```
  - Kernel (e.g. `uname -a`):
   ```
   ubuntu@Kube-Master01:~$ uname -a
   Linux Kube-Master01 5.4.0-1045-aws #47~18.04.1-Ubuntu SMP Tue Apr 13 15:58:14 UTC 2021 x86_64 x86_64 x86_64 
   GNU/Linux
   ```
  - Go version (e.g. `go version`):
1.16.6
  - Others:

  </details>
- <details><summary>Edge nodes Environment:</summary>
   Not relevant.
  - edgecore version (e.g. `edgecore --version`):
  - Hardware configuration (e.g. `lscpu`):
  - OS (e.g. `cat /etc/os-release`):
  - Kernel (e.g. `uname -a`):
  - Go version (e.g. `go version`):
  - Others:

  </details>

**Comment by BenAsaf** [comment]
Adding some info:
On my Desktop PC all worked fine. I guess there is an issue with EC2?
Any clue?

**Comment by SchmidtJonathan1** [comment]
Hello, got the same problem on my two virtual machines (CentOS 8 and Ubuntu 20.4).

**Comment by BenAsaf** [comment]
> Hello, got the same problem on my two virtual machines (CentOS 8 and Ubuntu 20.4).

I didn't continue to try fixing it.
Hopefully someone can shed some light.

**Comment by SchmidtJonathan1** [comment]
i figured it out.. Kubernetes version 1.22 released, if you use kubernetes 1.21 everthing works..

**Comment by slomo28** [comment]
I encountered the same problem, I used V1.21.3 at the beginning, it worked fine, the second time I used V1.22.0, I encountered this problem, I switched back to the old version, the problem was solved.
